src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -384,6 +384,44 @@ public static void LogEventHubsDependency(this ILogger logger, string namespaceN
             logger.LogInformation(DependencyFormat, "Azure Event Hubs", namespaceName, eventHubName, duration, startTime.ToString(CultureInfo.InvariantCulture), isSuccessful, context);
         }
 
+        /// <summary>
+        ///     Logs an Azure Iot Hub Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="namespaceName">Namespace of the resource</param>
+        /// <param name="iotHubName">Name of the IoT Hub resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="measurement">Measuring the latency to call the dependency</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogIotHubDependency(this ILogger logger, string namespaceName, string iotHubName, bool isSuccessful, DependencyMeasurement measurement, Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger));
+            Guard.NotNullOrWhitespace(iotHubName, nameof(iotHubName));
+            Guard.NotNullOrWhitespace(namespaceName, nameof(namespaceName));
+
+            LogIotHubDependency(logger, namespaceName, iotHubName, isSuccessful, measurement.StartTime, measurement.Elapsed, context);
+        }
+
+        /// <summary>
+        ///     Logs an Azure Iot Hub Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="namespaceName">Namespace of the resource</param>
+        /// <param name="iotHubName">Name of the Event Hub resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="startTime">Point in time when the interaction with the dependency was started</param>
+        /// <param name="duration">Duration of the operation</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogIotHubDependency(this ILogger logger, string namespaceName, string iotHubName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
```

Comment:
Azure IoT Hub is a per-instance service and doesn't have notion of a namespace
```suggestion
        public static void LogIotHubDependency(this ILogger logger, string iotHubName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
```